server/pulp/server/webservices/views/consumers.py

```diff
@@ -0,0 +1,1082 @@
+        :param request: WSGI request object
+        :type request: django.core.handlers.wsgi.WSGIRequest
+        :param consumer_id: The consumer ID.
+        :type consumer_id: str
+        :param schedule_id: the schedule id
+        :type schedule_id: str
+
+        :raises MissingResource: if consumer/schedule does not exist
+
+        :return: Response containing consumer's schedule <action>
+        :rtype: django.http.HttpResponse
+        """
+
+        scheduled_call = None
+        for call in self.manager.get(consumer_id, self.ACTION):
+            if call.id == schedule_id:
+                scheduled_call = call
+                break
+        if scheduled_call is None:
+            raise MissingResource(consumer_id=consumer_id, schedule_id=schedule_id)
+
+        scheduled_obj = scheduled_unit_management_obj(scheduled_call.for_display())
+        add_link_schedule(scheduled_obj, self.ACTION, consumer_id)
+        return generate_json_response_with_pulp_encoder(scheduled_obj)
+
+    @auth_required(authorization.UPDATE)
+    @json_body_allow_empty
+    def put(self, request, consumer_id,  schedule_id):
+        """
+        Update a specific schedule <action>.
+
+        :param request: WSGI request object
+        :type request: django.core.handlers.wsgi.WSGIRequest
+        :param consumer_id: The consumer ID.
+        :type consumer_id: str
+        :param schedule_id: the schedule id
+        :type schedule_id: str
+
+        :return: Response containing consumer's updated schedule <action>
+        :rtype: django.http.HttpResponse
+        """
+
+        schedule_data = request.body_as_json
+        options = schedule_data.pop('options', None)
+        units = schedule_data.pop('units', None)
+
+        if 'schedule' in schedule_data:
+            schedule_data['iso_schedule'] = schedule_data.pop('schedule')
+
+        schedule = self.manager.update_schedule(consumer_id, schedule_id, units,
+                                                options, schedule_data)
+
+        scheduled_obj = scheduled_unit_management_obj(schedule.for_display())
+        add_link_schedule(scheduled_obj, self.ACTION, consumer_id)
+        return generate_json_response_with_pulp_encoder(scheduled_obj)
+
+    @auth_required(authorization.DELETE)
+    def delete(self, request, consumer_id, schedule_id):
+        """
+        Delete a specific schedule <action>.
+
+        :param request: WSGI request object
+        :type request: django.core.handlers.wsgi.WSGIRequest
+        :param consumer_id: The consumer ID.
+        :type consumer_id: str
+        :param schedule_id: the schedule id
+        :type schedule_id: str
+
+        :return: An empty response
+        :rtype: django.http.HttpResponse
+        """
+
+        self.manager.delete_schedule(consumer_id, schedule_id)
+        return generate_json_response(None)
+
+
+class UnitInstallScheduleView(ConsumerUnitActionScheduleView):
+    """
+    View for scheduled content install on the consumer.
```

Comment:
These class names and doc block should be changed if you take my suggestions for their parents.